lib/pbench/test/unit/agent/task/test_tools_kill.py

```diff
@@ -0,0 +1,440 @@
+                for name, val in self.dir_names:
+                    yield AnotherPath(name, val)
+
+        pr = FakePbenchRun(
+            [
+                ("abc.file", None),  # top level file
+                ("filenotf", None),  # directory with no .uuid file
+                ("run1", "abcdef"),  # directory with a file
+                ("run2", "ghijkl"),  # directory with a file
+            ]
+        )
+        pairs = list(kill.gen_run_directories(pr))
+        assert pairs[0][0].name == "run1/tm" and pairs[0][1] == "abcdef"
+        assert pairs[1][0].name == "run2/tm" and pairs[1][1] == "ghijkl"
+
+    @staticmethod
+    def test_gen_host_names():
+        class MockToolGroup:
+            """A simple ToolGroup mock to generate host names from different
+            tool group directories.
+            """
+
+            TOOL_GROUP_PREFIX = "tests-v9"
+
+            _named_sets = {
+                "light": {
+                    "remote1.example.com": None,
+                    "localhost.example.com": None,
+                },
+                "medium": {
+                    "remote2.example.com": None,
+                    "remote3.example.com": None,
+                },
+                "heavy": {"localhost.example.com": None},
+            }
+
+            _expected_hosts = set(
+                ("remote1.example.com", "remote2.example.com", "remote3.example.com")
+            )
+
+            def __init__(self, name: str, run_dir: Any):
+                self.hostnames = self._named_sets[name]
+
+        class MockPathGlob:
+            """Mock of a Path object to direct the `.glob()` method's behavior."""
+
+            def __init__(self, name: str):
+                self.name = name
+
+            def glob(self, prefix: str):
+                if self.name == "no-tool-group-dirs":
+                    # The special name which for a directory which "does not
+                    # have tool groups".
+                    return
+                for n in MockToolGroup._named_sets.keys():
+                    # Return the 3 tools groups that are found with the prefix
+                    # attached.
+                    yield AnotherPath(f"{prefix[:-2]}-{n}")
+
+        class MockPathForGlob:
+            """Mock of a Path object just to generate a parent object which
+            uses the MockPathGlob class.  We don't need a name for the object
+            itself, just need to hand out the parent when referenced.
+            """
+
+            def __init__(self, parent: str):
+                self.parent = MockPathGlob(parent)
+
+        class MockLocalRemoteHost:
+            """A simple mock for the LocalRemoteHost behaviors."""
+
+            def is_local(self, host_name: str):
+                return host_name.startswith("localhost")
+
+        # To verify the case where there are no tool groups created, we make a
+        # special Path object which will behavior like it has no tool groups.
```

Comment:
"... behave as if..."?

Reply:
Or "will expose no tool groups"? The "behave as if" somehow implies it does have tool groups, but lies. 👀